### PR TITLE
RichText Native: Adds onEnter callback and function placeholder.

### PR DESCRIPTION
**Pull request by diegoreymendez**
## Description

This PR adds an onEnter callback property to the native implementation of RichText.

## How has this been tested?

Tested in `gutenberg-mobile`.  There's no logic yet but I tested the callback is being called properly.

## Types of changes

Adds an `onEnter` callback property to the native implementation of RichText.
Adds a placeholder method and binds it to the callback.
Adds a console log that's printed whenever the callback is executed.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by SergioEstevao** [comment]
@diegoreymendez don't you need a branch/PR on gutenberg-mobile that refers to this and to the updated version of react-aztec repo?

**Comment by diegoreymendez** [comment]
@SergioEstevao - You're right.  The PR is here: https://github.com/wordpress-mobile/gutenberg-mobile/pull/136

File: packages/editor/src/components/rich-text/index.native.js [hunk 1]
```diff
@@ -94,6 +95,10 @@ export class RichText extends Component {
 		);
 	}
 
+	onEnter( event ) {
```
Comment (SergioEstevao) [inline packages/editor/src/components/rich-text/index.native.js]: You need to remove the parameter here, now that you update the onEnter interface on the RN-Aztec repo.

**Review by SergioEstevao — CHANGES_REQUESTED** [review]
Just need to update to the latest API for the onEnter event.

**Comment by diegoreymendez** [comment]
@SergioEstevao - My bad... I had made the changes in the submodule but failed to push them properly.  It's not updated.

**Comment by diegoreymendez** [comment]
@SergioEstevao - This is ready for another look.

**Review by SergioEstevao — APPROVED** [review]
Good work.

**Pull request merged**